### ISSUE TITLE
False negative when unpacking a tuple as a mapping

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#1108]
```py
xs = {}
ys = ()
zs = {**xs, **ys}
```
This raises `TypeError: 'tuple' object is not a mapping`, but Pyright thinks it's fine.
Turns out that `ys` has to be a mapping, which is different from `dict.update`.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.80, which I just published. It will also be in the next release of pylance.
